[Bug]: panic: runtime error: invalid memory address or nil pointer dereference (calculateContinuousModeFees)

### Problem encountered

vega-market-sim fuzz testing panics

[vega-sim-ok5y9kwa.zip](https://github.com/user-attachments/files/16613161/vega-sim-ok5y9kwa.zip)

```
2024-08-14T13:26:08.730+0100	INFO	core	node/node.go:99	starting vega	{"version": "v0.78.0-dev", "commit-hash": "4815e1348eb10cd81e237a5886be3393b2665594"}
```
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x105d35de4]

goroutine 76489 [running]:
code.vegaprotocol.io/vega/libs/num.(*Uint).Mul(...)
	/Users/caed/Projects/vega-market-sim/extern/vega/libs/num/uint.go:347
code.vegaprotocol.io/vega/core/fee.(*Engine).calculateContinuousModeFees(0x140023a7e00, 0x0?)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/fee/engine.go:766 +0x54
code.vegaprotocol.io/vega/core/fee.(*Engine).calculateAuctionModeFees(0x14005f45a78?, 0x104d0cb30?)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/fee/engine.go:783 +0x20
code.vegaprotocol.io/vega/core/fee.(*Engine).getAuctionModeFeesAndTransfers(0x140023a7e00, 0x140085984a8, {0x1091c8350, 0x14000e207e0}, {0x1091aa2c0, 0x14000df3a70}, {0x1091aa2e0, 0x14000df3b60})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/fee/engine.go:733 +0x44
code.vegaprotocol.io/vega/core/fee.(*Engine).CalculateForAuctionMode(0x20?, {0x140085984a0, 0x1, 0x14009936b40?}, {0x1091c8350, 0x14000e207e0}, {0x1091aa2c0, 0x14000df3a70}, {0x1091aa2e0, 0x14000df3b60})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/fee/engine.go:356 +0x128
code.vegaprotocol.io/vega/core/execution/spot.(*Market).calculateFeesForTrades(0x1400258f800, {0x140085984a0, 0x1, 0x1})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/spot/market.go:3186 +0x88
code.vegaprotocol.io/vega/core/execution/spot.(*Market).calculateFees(0x1400258f800, {0x14009299ec0, 0x40}, 0xe9adf, 0x0, 0x1)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/spot/market.go:3164 +0xcc
code.vegaprotocol.io/vega/core/execution/spot.(*Market).validateOrderAmendment(0x1400258f800, 0x1400302fee0, 0x14004b4a7e0)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/spot/market.go:2384 +0x3c4
code.vegaprotocol.io/vega/core/execution/spot.(*Market).amendOrder(0x1400258f800, {0x1091d1e58, 0x140072a7c50}, 0x14004b4a7e0, {0x1400bfe81c0, 0x40})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/spot/market.go:2090 +0x170
code.vegaprotocol.io/vega/core/execution/spot.(*Market).AmendOrderWithIDGenerator(0x1400258f800, {0x1091d1e58?, 0x140072a7c50}, 0x40?, {0x1400bfe81c0?, 0x104c7df94?}, {0x1091a0800?, 0x1400c870f18?})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/spot/market.go:2025 +0x100
code.vegaprotocol.io/vega/core/execution.(*Engine).AmendOrder(0x14000cd3c00, {0x1091d1e58, 0x140072a7c50}, 0x14004b4a7e0, {0x1400bfe81c0, 0x40}, {0x1091a0800, 0x1400c870f18})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/engine.go:1009 +0x2a8
code.vegaprotocol.io/vega/core/processor.(*App).DeliverAmendOrder(0x14000972700, {0x1091d1e58, 0x140072a7c50}, {0x1091ff150, 0x14002fd6140}, {0x1400db5f300, 0x40})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:2118 +0x200
code.vegaprotocol.io/vega/core/processor.NewApp.addDeterministicID.func25({0x1091d1e58, 0x140072a7c50}, {0x1091ff150, 0x14002fd6140})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:595 +0x64
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).SendTransactionResult.func26({0x1091d1e58, 0x140072a7c50}, {0x1091ff150?, 0x14002fd6140?})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:658 +0x4c
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0x140006c8a80, {0x108f6b060?, 0x1?}, 0x14002fd3740)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:179 +0x490
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0x1400b857878?, {0x1091d1e20?, 0x10b93f8a0?}, 0x10b93f8a0?)
	/Users/caed/Projects/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x30
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x1400013d680)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:212 +0x488
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x1400013d680, 0x3b9aca00)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:289 +0x194
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime in goroutine 76487
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1e0
```

### Observed behaviour

panic

### Expected behaviour

no panic

### Steps to reproduce

```bash
replay
```


### Software version

4815e1348eb10cd81e237a5886be3393b2665594

### Failing test

vega-market-sim fuzz test

### Jenkins run

_No response_

### Configuration used

```shell
{
  "genesis_time": "2023-05-09T10:00:00.000000000Z",
  "chain_id": "CUSTOM",
  "initial_height": "0",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": ["ed25519"]
    },
    "version": {}
  },
  "validators": [
    {
      "address": "7DFD3744A773C166B36F170E4F5EF3E998703F4E",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "2lPO6I0p5uCBZzeoA2+ECJ2+57LtLVMjXpxH5za9TvE="
      },
      "power": "10",
      "name": ""
    }
  ],
  "app_hash": "",
  "app_state": {
    "assets": {
      "0000000000000000000000000000000000000000000000000000000000000000": {
        "name": "VOTE",
        "symbol": "VOTE",
        "decimals": 5,
        "quantum": "1",
        "min_lp_stake": "1",
        "source": {
          "builtin_asset": {
            "max_faucet_amount_mint": "100000000000000000"
          }
        }
      }
    },
    "validators": {
      "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=": {
        "id": "c2b0ab69fb1bd19862c04df401920ea62d3e0bce41a50148240f42017824d17a",
        "vega_pub_key": "dfcf0fa8d67cfa11b3a511e320f3c47807a5e0735cc06dd9c2bc00f7ccc67117",
        "vega_pub_key_index": 1,
        "ethereum_address": "0x9e5BEEC6E56B28cCbd02864840B0f1e0125e42Ce",
        "tm_pub_key": "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=",
        "info_url": "",
        "country": "",
        "name": "",
        "avatar_url": ""
      }
    },
    "network": {
      "replay_attack_threshold": 150
    },
    "network_parameters": {
      "blockchains.ethereumConfig": "{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"staking_bridge_contract\": { \"address\": \"0xfce2CC92203A266a9C8e67461ae5067c78f67235\", \"deployment_block_height\": 11001702}, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }",
      "blockchains.evmBridgeConfigs": "{\"configs\":[{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }]}",
      "blockchains.ethereumRpcAndEvmCompatDataSourcesConfig": "{\"configs\": []}",
      "governance.proposal.asset.maxClose": "8760h0m0s",
      "governance.proposal.asset.maxEnact": "8760h0m0s",
      "governance.proposal.asset.minClose": "1s",
      "governance.proposal.asset.minEnact": "1s",
      "governance.proposal.asset.minProposerBalance": "1",
      "governance.proposal.asset.minVoterBalance": "1",
      "governance.proposal.asset.requiredMajority": "0.66",
      "governance.proposal.asset.requiredParticipation": "0.00001",
      "governance.proposal.freeform.maxClose": "8760h0m0s",
      "governance.proposal.freeform.minClose": "1s",
      "governance.proposal.freeform.minProposerBalance": "1",
      "governance.proposal.freeform.minVoterBalance": "1",
      "governance.proposal.freeform.requiredMajority": "0.66",
      "governance.proposal.freeform.requiredParticipation": "0.00001",
      "governance.proposal.market.maxClose": "8760h0m0s",
      "governance.proposal.market.maxEnact": "8760h0m0s",
      "governance.proposal.market.minClose": "1s",
      "governance.proposal.market.minEnact": "2s",
      "governance.proposal.market.minProposerBalance": "1",
      "governance.proposal.market.minVoterBalance": "1",
      "governance.proposal.market.requiredMajority": "0.66",
      "governance.proposal.market.requiredParticipation": "0.00001",
      "governance.proposal.updateAsset.maxClose": "720h",
      "governance.proposal.updateAsset.maxEnact": "720h",
      "governance.proposal.updateAsset.minClose": "1s",
      "governance.proposal.updateAsset.minEnact": "1s",
      "governance.proposal.updateAsset.minProposerBalance": "1",
      "governance.proposal.updateAsset.minVoterBalance": "1",
      "governance.proposal.updateAsset.requiredMajority": "0.66",
      "governance.proposal.updateAsset.requiredParticipation": "0.09",
      "governance.proposal.updateMarket.maxClose": "8760h0m0s",
      "governance.proposal.updateMarket.maxEnact": "8760h0m0s",
      "governance.proposal.updateMarket.minClose": "1s",
      "governance.proposal.updateMarket.minEnact": "1s",
      "governance.proposal.updateMarket.minProposerBalance": "1",
      "governance.proposal.updateMarket.minProposerEquityLikeShare": "0",
      "governance.proposal.updateMarket.minVoterBalance": "1",
      "governance.proposal.updateMarket.requiredMajority": "0.66",
      "governance.proposal.updateMarket.requiredMajorityLP": "0.66",
      "governance.proposal.updateMarket.requiredParticipation": "0.00001",
      "governance.proposal.updateMarket.requiredParticipationLP": "0",
      "governance.proposal.updateNetParam.maxClose": "8760h0m0s",
      "governance.proposal.updateNetParam.maxEnact": "8760h0m0s",
      "governance.proposal.updateNetParam.minClose": "1s",
      "governance.proposal.updateNetParam.minEnact": "1s",
      "governance.proposal.updateNetParam.minProposerBalance": "1",
      "governance.proposal.updateNetParam.minVoterBalance": "1",
      "governance.proposal.updateNetParam.requiredMajority": "0.5",
      "governance.proposal.updateNetParam.requiredParticipation": "0.00001",
      "governance.proposal.referralProgram.maxClose": "8760h0m0s",
      "governance.proposal.referralProgram.maxEnact": "8760h0m0s",
      "governance.proposal.referralProgram.minClose": "1s",
      "governance.proposal.referralProgram.minEnact": "1s",
      "governance.proposal.referralProgram.minProposerBalance": "1",
      "governance.proposal.referralProgram.minVoterBalance": "1",
      "governance.proposal.referralProgram.requiredMajority": "0.5",
      "governance.proposal.referralProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeDiscountProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.minClose": "1s",
      "governance.proposal.VolumeDiscountProgram.minEnact": "1s",
      "governance.proposal.VolumeDiscountProgram.minProposerBalance": "1",
      "governance.proposal.VolumeDiscountProgram.minVoterBalance": "1",
      "governance.proposal.VolumeDiscountProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeDiscountProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeRebateProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeRebateProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeRebateProgram.minClose": "1s",
      "governance.proposal.VolumeRebateProgram.minEnact": "1s",
      "governance.proposal.VolumeRebateProgram.minProposerBalance": "1",
      "governance.proposal.VolumeRebateProgram.minVoterBalance": "1",
      "governance.proposal.VolumeRebateProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeRebateProgram.requiredParticipation": "0.00001",
      "governance.proposal.transfer.maxClose": "8760h0m0s",
      "governance.proposal.transfer.maxEnact": "8760h0m0s",
      "governance.proposal.transfer.minClose": "1s",
      "governance.proposal.transfer.minEnact": "1s",
      "governance.proposal.transfer.minProposerBalance": "1",
      "governance.proposal.transfer.minVoterBalance": "1",
      "governance.proposal.transfer.requiredMajority": "0.5",
      "governance.proposal.transfer.requiredParticipation": "0.00001",
      "limits.markets.maxPeggedOrders": "1500",
      "limits.markets.proposePerpetualEnabled": "1",
      "limits.markets.ammPoolEnabled": "1",
      "market.auction.maximumDuration": "168h",
      "market.auction.minimumDuration": "1s",
      "market.fee.factors.infrastructureFee": "0.0005",
      "market.fee.factors.makerFee": "0.0002",
      "market.fee.factors.treasuryFee": "0.0001",
      "market.fee.factors.buybackFee": "0.0001",
      "market.liquidity.bondPenaltyParameter": "0.1",
      "market.liquidity.earlyExitPenalty": "0.1",
      "market.liquidity.maximumLiquidityFeeFactorLevel": "0.03",
      "market.liquidity.minimum.probabilityOfTrading.lpOrders": "1e-6",
      "market.liquidity.probabilityOfTrading.tau.scaling": "10.0",
      "market.liquidity.providersFeeCalculationTimeStep": "1s",
      "market.liquidity.sla.nonPerformanceBondPenaltyMax": "0.5",
      "market.liquidity.sla.nonPerformanceBondPenaltySlope": "2",
      "market.liquidity.stakeToCcyVolume": "1.0",
      "market.liquidity.targetstake.triggering.ratio": "0.25",
      "market.liquidityProvision.minLpStakeQuantumMultiple": "5000",
      "market.liquidityProvision.shapes.maxSize": "100",
      "market.margin.scalingFactors": "{\"search_level\": 1.1, \"initial_margin\": 1.5, \"collateral_release\": 1.7}",
      "market.monitor.price.defaultParameters": "{\"triggers\": [{\"auction_extension\": 300, \"horizon\": 43200, \"probability\": \"0.9999999\"}] }",
      "market.stake.target.scalingFactor": "0.0001",
      "market.stake.target.timeWindow": "1h",
      "market.value.windowLength": "2h0m0s",
      "network.checkpoint.timeElapsedBetweenCheckpoints": "5m",
      "network.floatingPointUpdates.delay": "5m",
      "network.markPriceUpdateMaximumFrequency": "5s",
      "network.transaction.defaultgas": "1",
      "network.transactions.maxgasperblock": "3000000",
      "network.transactions.minBlockCapacity": "32",
      "network.validators.ersatz.multipleOfTendermintValidators": "0",
      "network.validators.ersatz.rewardFactor": "0.85",
      "network.validators.incumbentBonus": "0.05",
      "network.validators.minimumEthereumEventsForNewValidator": "3",
      "network.validators.multisig.numberOfSigners": "13",
      "network.validators.tendermint.number": "13",
      "reward.asset": "0000000000000000000000000000000000000000000000000000000000000000",
      "reward.staking.delegation.competitionLevel": "1.3",
      "reward.staking.delegation.delegatorShare": "0.8",
      "reward.staking.delegation.maxPayoutPerEpoch": "9890000000000000000000",
      "reward.staking.delegation.maxPayoutPerParticipant": "700000000000000000000",
      "reward.staking.delegation.minimumValidatorStake": "3000000000000000000000",
      "reward.staking.delegation.minValidators": "5",
      "reward.staking.delegation.optimalStakeMultiplier": "3.0",
      "reward.staking.delegation.payoutDelay": "0h",
      "reward.staking.delegation.payoutFraction": "1.0",
      "rewards.marketCreationQuantumMultiple": "1",
      "snapshot.interval.length": "300",
      "spam.pow.difficulty": "15",
      "spam.pow.hashFunction": "sha3_24_rounds",
      "spam.pow.increaseDifficulty": "0",
      "spam.pow.numberOfPastBlocks": "100",
      "spam.pow.numberOfTxPerBlock": "1000",
      "spam.protection.delegation.min.tokens": "100000000000000000",
      "spam.protection.max.batchSize": "30",
      "spam.protection.max.delegations": "360",
      "spam.protection.max.proposals": "300",
      "spam.protection.max.votes": "300",
      "spam.protection.maxUserTransfersPerEpoch": "10000",
      "spam.protection.minimumWithdrawalQuantumMultiple": "10",
      "spam.protection.minMultisigUpdates": "100000000000000000000",
      "spam.protection.proposal.min.tokens": "2000000000000000000000",
      "spam.protection.voting.min.tokens": "1000000000000000000",
      "spam.protection.applyReferral.min.funds": "0",
      "transfer.fee.factor": "0.001",
      "transfer.minTransferQuantumMultiple": "0",
      "transfer.fee.maxQuantumAmount": "100",
      "transfer.feeDiscountMinimumTrackedAmount": "0.001",
      "transfer.feeDiscountDecayFraction": "0.5",
      "validator.performance.scaling.factor": "0",
      "validators.delegation.minAmount": "1",
      "validators.epoch.length": "2m",
      "validators.vote.required": "0.67",
      "referralProgram.maxReferralTiers": "10",
      "referralProgram.maxReferralRewardFactor": "1",
      "referralProgram.maxReferralDiscountFactor": "1",
      "referralProgram.maxPartyNotionalVolumeByQuantumPerEpoch": "10000",
      "referralProgram.minStakedVegaTokens": "0",
      "referralProgram.maxReferralRewardProportion": "1",
      "market.aggressiveOrderBlockDelay": "1",
      "market.amm.minCommitmentQuantum": "1",
      "market.liquidity.maxAmmCalculationLevels": "100"
    },
    "network_limits": {
      "propose_market_enabled": true,
      "propose_asset_enabled": true,
      "bootstrap_block_count": 0
    },
    "checkpoint": {
      "load_hash": ""
    }
  }
}
```


### Relevant log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x105d35de4]

goroutine 76489 [running]:
code.vegaprotocol.io/vega/libs/num.(*Uint).Mul(...)
	/Users/caed/Projects/vega-market-sim/extern/vega/libs/num/uint.go:347
code.vegaprotocol.io/vega/core/fee.(*Engine).calculateContinuousModeFees(0x140023a7e00, 0x0?)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/fee/engine.go:766 +0x54
code.vegaprotocol.io/vega/core/fee.(*Engine).calculateAuctionModeFees(0x14005f45a78?, 0x104d0cb30?)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/fee/engine.go:783 +0x20
code.vegaprotocol.io/vega/core/fee.(*Engine).getAuctionModeFeesAndTransfers(0x140023a7e00, 0x140085984a8, {0x1091c8350, 0x14000e207e0}, {0x1091aa2c0, 0x14000df3a70}, {0x1091aa2e0, 0x14000df3b60})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/fee/engine.go:733 +0x44
code.vegaprotocol.io/vega/core/fee.(*Engine).CalculateForAuctionMode(0x20?, {0x140085984a0, 0x1, 0x14009936b40?}, {0x1091c8350, 0x14000e207e0}, {0x1091aa2c0, 0x14000df3a70}, {0x1091aa2e0, 0x14000df3b60})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/fee/engine.go:356 +0x128
code.vegaprotocol.io/vega/core/execution/spot.(*Market).calculateFeesForTrades(0x1400258f800, {0x140085984a0, 0x1, 0x1})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/spot/market.go:3186 +0x88
code.vegaprotocol.io/vega/core/execution/spot.(*Market).calculateFees(0x1400258f800, {0x14009299ec0, 0x40}, 0xe9adf, 0x0, 0x1)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/spot/market.go:3164 +0xcc
code.vegaprotocol.io/vega/core/execution/spot.(*Market).validateOrderAmendment(0x1400258f800, 0x1400302fee0, 0x14004b4a7e0)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/spot/market.go:2384 +0x3c4
code.vegaprotocol.io/vega/core/execution/spot.(*Market).amendOrder(0x1400258f800, {0x1091d1e58, 0x140072a7c50}, 0x14004b4a7e0, {0x1400bfe81c0, 0x40})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/spot/market.go:2090 +0x170
code.vegaprotocol.io/vega/core/execution/spot.(*Market).AmendOrderWithIDGenerator(0x1400258f800, {0x1091d1e58?, 0x140072a7c50}, 0x40?, {0x1400bfe81c0?, 0x104c7df94?}, {0x1091a0800?, 0x1400c870f18?})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/spot/market.go:2025 +0x100
code.vegaprotocol.io/vega/core/execution.(*Engine).AmendOrder(0x14000cd3c00, {0x1091d1e58, 0x140072a7c50}, 0x14004b4a7e0, {0x1400bfe81c0, 0x40}, {0x1091a0800, 0x1400c870f18})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/execution/engine.go:1009 +0x2a8
code.vegaprotocol.io/vega/core/processor.(*App).DeliverAmendOrder(0x14000972700, {0x1091d1e58, 0x140072a7c50}, {0x1091ff150, 0x14002fd6140}, {0x1400db5f300, 0x40})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:2118 +0x200
code.vegaprotocol.io/vega/core/processor.NewApp.addDeterministicID.func25({0x1091d1e58, 0x140072a7c50}, {0x1091ff150, 0x14002fd6140})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:595 +0x64
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).SendTransactionResult.func26({0x1091d1e58, 0x140072a7c50}, {0x1091ff150?, 0x14002fd6140?})
	/Users/caed/Projects/vega-market-sim/extern/vega/core/processor/abci.go:658 +0x4c
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0x140006c8a80, {0x108f6b060?, 0x1?}, 0x14002fd3740)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:179 +0x490
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0x1400b857878?, {0x1091d1e20?, 0x10b93f8a0?}, 0x10b93f8a0?)
	/Users/caed/Projects/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x30
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x1400013d680)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:212 +0x488
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x1400013d680, 0x3b9aca00)
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:289 +0x194
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime in goroutine 76487
	/Users/caed/Projects/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1e0
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,9 +22,9 @@
 - [11535](https://github.com/vegaprotocol/vega/issues/11535) - Added support for lottery rank distribution strategy.
 - [11536](https://github.com/vegaprotocol/vega/issues/11536) - Make the batch market instructions errors programmatically usable.
 - [11546](https://github.com/vegaprotocol/vega/issues/11546) - Add validation to market proposals metadata.
-- [11562](https://github.com/vegaprotocol/vega/issues/11562) - Update average notional metric with mark price at the end of the epoch and when calculating live score. 
+- [11562](https://github.com/vegaprotocol/vega/issues/11562) - Update average notional metric with mark price at the end of the epoch and when calculating live score.
 - [11570](https://github.com/vegaprotocol/vega/issues/11570) - Include the required set of parties for evaluation for eligible entities reward.
-  
+
 ### 🐛 Fixes
 
 - [11521](https://github.com/vegaprotocol/vega/issues/11521) - Restore `AMM` position factor when loading from a snapshot.
@@ -35,6 +35,7 @@
 - [11540](https://github.com/vegaprotocol/vega/issues/11540) - Fix spam check for spots to use not double count quantum.
 - [11542](https://github.com/vegaprotocol/vega/issues/11542) - Fix non determinism in lottery ranking.
 - [11544](https://github.com/vegaprotocol/vega/issues/11544) - Fix empty candles stream.
+- [11579](https://github.com/vegaprotocol/vega/issues/11579) - Spot calculate fee on amend, use order price if no amended price is provided.
 
 
 ## 0.77.5

--- a/core/execution/spot/market.go
+++ b/core/execution/spot/market.go
@@ -2380,12 +2380,12 @@ func (m *Market) validateOrderAmendment(order *types.Order, amendment *types.Ord
 		existingHoldingQty, existingHoldingFee := m.orderHoldingTracker.GetCurrentHolding(order.ID)
 		oldHoldingRequirement := num.Sum(existingHoldingQty, existingHoldingFee)
 		newFeesRequirement := num.UintZero()
-		if m.as.InAuction() {
-			newFeesRequirement, _ = m.calculateFees(order.Party, remaining, amendment.Price, order.Side)
-		}
 		price := order.Price
 		if amendment.Price != nil {
 			price, _ = num.UintFromDecimal(amendment.Price.ToDecimal().Mul(m.priceFactor))
+		}
+		if m.as.InAuction() {
+			newFeesRequirement, _ = m.calculateFees(order.Party, remaining, price, order.Side)
 		}
 		if order.PeggedOrder != nil {
 			p, err := m.getNewPeggedPrice(order)

